packages/cli/commands/validation/themeValidators/templateLimitValidator.js

```diff
@@ -0,0 +1,27 @@
+const { getDirectoryFiles } = require('@hubspot/cli-lib/lib/walk');
+const { ERROR_SEVERITY } = require('../validationErrorUtils');
+
+const VALIDATOR_NAME = 'TemplateLimitValidator';
+const TEMPLATE_LIMIT = 50;
+
+// TODO branden are we checking for all files here? OR should we check for templates specifically
+
+// Validate that the theme does not contain more than TEMPLATE_LIMIT templates
+function templateLimitValidator(absoluteThemePath) {
+  return getDirectoryFiles(absoluteThemePath).then(files => {
```

Comment:
Since templates could be nested at any depth, I think we need to walk here and filter out HTML files (we might even need to filter out partials)

Reply:
Oh good catch. I put this in the wrong place after testing. I meant to put it in the `ThemeConfigValidator`.

Reply:
> we might even need to filter out partials

@KatzMitch should we be filtering out partials? Also I'm currently just checking for total file count (not specifically looking for templates) and I think that's wrong.

Reply:
Yeah it's 50 top-level templates, excludes partials

Reply:
@gcorne @anthmatic is there a good way to differentiate between templates and partials?

Reply:
Nothing that exists now, I think this would require reading the file and parsing out the HTML comment (annotation) at the top.